application/event_processor/event_processor.py

```diff
@@ -125,24 +125,43 @@ def lambda_handler(event: SQSEvent, context: LambdaContext) -> None:
     for env_var in EXPECTED_ENVIRONMENT_VARIABLES:
         if env_var not in environ:
             logger.error(f"Environmental variable {env_var} not present")
+            return
 
     counter = 0
     for record in event.records:
         counter += 1
         if counter > 1:
-            raise Exception("More than one record found in event")
+            logger.error("More than one record found in event", extra={"event": event})
+            return
 
-    message = next(event.records).body
-    change_event = extract_message(message)
+    record = next(event.records)
+    message = record.body
+    change_event = extract_body(message)
+    sequence_number = get_sequence_number(record)
+    db_latest_sequence_number = get_latest_sequence_id_for_a_given_odscode_from_dynamodb(change_event["ODSCode"])
+    if int(0 if sequence_number is None else sequence_number) < int(
+        0 if db_latest_sequence_number is None else db_latest_sequence_number
+    ):
+        logger.error("Sequence id is smaller than the existing one in db for a given odscode, so will be ignored")
+        return
+    sqs_timestamp = str(record.attributes["SentTimestamp"])
+    # Save Event to dynamo so can be retrieved later
+    add_change_request_to_dynamodb(change_event, sequence_number, sqs_timestamp)
     logger.info(f"Attempting to validate change_event: {change_event}")
-    validate_event(change_event)
-
+    if sequence_number is None:
```

Comment:
should move this above your new sequence number checker otherwise you will return because sequence id is smaller rather than empy as you set empty as 0 above which could be less than a previous one